### ISSUE TITLE
Conky positioning broken

### ISSUE THREAD

**Issue by SethDusek** [linuxmint/Cinnamon#5566]
So, my conky (and every other conky) isn't being positioned properly. I originally filed this issue on the conky page, but I realized that conky was being positioned just fine with openbox. So it seems to be a cinnamon specific problem. The conky widgets are all ending up on the top left

Original issue: https://github.com/brndnmtthws/conky/issues/228

**Comment by leigh123linux** [comment]
Works fine here

![screenshot from 2016-08-18 16-05-32](https://cloud.githubusercontent.com/assets/1299302/17778921/b190c700-655d-11e6-8e86-df9bf71b5474.png)

**Comment by SethDusek** [comment]
@leigh123linux Could you post your conkyrc? I could show you where it ends up on my system

**Comment by leigh123linux** [comment]
Here's my conkyrc

```
# **********************************************************************
# "CPU Panel (8-core)" theme for Conky by Tony George (teejee2008@gmail.com)
#
# Webpage: http://teejeetech.blogspot.in/
# **********************************************************************


background yes
double_buffer yes

alignment top_right

border_width 1
cpu_avg_samples 2
default_color white
default_outline_color white
default_shade_color white
draw_borders no
draw_graph_borders yes
draw_outline no
draw_shades no

gap_x 90
gap_y 60
net_avg_samples 2
no_buffers yes
out_to_console no
out_to_stderr no
extra_newline no

own_window yes
own_window_type desktop
own_window_transparent yes
own_window_colour 000000
own_window_argb_visual no
own_window_argb_value 0
own_window_hints undecorated,below,sticky,skip_taskbar,skip_pager

minimum_size 170 0
stippled_borders 0
update_interval 2.0
uppercase no
use_spacer none

show_graph_scale no
show_graph_range no

use_xft yes
xftalpha 0.1
xftfont Droid Sans:size=10
color0 white
color1 EAEAEA
color2 FFA300
color3 grey

TEXT
${color2}CPU ${color0}${alignr}${cpu cpu0}%
${cpubar cpu0 5,}
${top name 1} $alignr ${top cpu 1}%
${top name 2} $alignr ${top cpu 2}%
${top name 3} $alignr ${top cpu 3}%
${top name 4} $alignr ${top cpu 4}%
${top name 5} $alignr ${top cpu 5}%

${color2}RAM ${color0}${alignr}${mem}
${membar 5,}
${top_mem name 1} $alignr ${top_mem mem_res 1}
${top_mem name 2} $alignr ${top_mem mem_res 2}
${top_mem name 3} $alignr ${top_mem mem_res 3}
${top_mem name 4} $alignr ${top_mem mem_res 4}
${top_mem name 5} $alignr ${top_mem mem_res 5}
```

**Comment by SethDusek** [comment]
@leigh123linux here is where it is stuck http://i.shibe.ml/QmNhggfPYxc4MUsatAQJg6mWBxMV6bPUFt2PFuqFbeac9D.png

**Comment by feren** [comment]
> stuck
Can't you move it with Alt + Drag, or is it completely stuck there?

**Comment by SethDusek** [comment]
@feren

Are you supposed to be able to alt+drag conky windows? If so, it's not dragging

**Comment by mainmachine** [comment]
ALT+click+drag... just tried and my mind was blown. Anyway, works on 3.0.7...

**Comment by feren** [comment]
@SethDusek @mainmachine Alt + Drag is...
1. In Linux Mint 18's release notes page under tips for moving windows on small screens
2. I think it's somewhere in the conky manual too

**Comment by feren** [comment]
@SethDusek 
What version of Linux Mint do you have and Cinnamon too? `cinnamon-settings info`

**Comment by SethDusek** [comment]
@feren mint 18 and cinnamon 3.0.7. This was happening on 17 too. I have no extensions enabled that could cause this, but there might be a setting i accidentally enabled or something, since nobody else seems to be having this issue

**Comment by feren** [comment]
Have you tried making another user (it's easier than trying a dconf Cinnamon reset instead) and putting the ConkyRC file on there and testing it on another user?

**Comment by SethDusek** [comment]
@feren Created a new user and it worked, conky is positioned just fine. So what should I do? Should I wipe my cinnamon settings? What exactly would I lose if I wiped the dconf settings?

**Comment by feren** [comment]
@SethDusek Try there whilst only running metacity (use `metacity --replace`)
```
gsettings reset-recursively org.cinnamon
cinnamon --reset
```

**Comment by feren** [comment]
@SethDusek 
You'd only lose the Panel and theme settings (including Applets currently in use), you'll have the background reset to default, etc

**Comment by mainmachine** [comment]
@SethDusek - Here's how you can back up your dconf db into a human readable format:
```
cd
cp ~/.config/dconf/user ~/.config/dconf/test
printf %s\\n "user-db:test" > db_profile
DCONF_PROFILE=~/db_profile dconf dump / > old_settings
```
You can edit this file manually, scan through it, grep it, whatever to look at/modify/extract your old settings, or you can even import it into a new one (thus breaking the new one :wink: ).

Source: http://unix.stackexchange.com/questions/199836/how-can-i-view-the-content-of-a-backup-of-the-dconf-database-file

**Comment by SethDusek** [comment]
Okay I fixed it. Turns out in org.cinnamon.desktop.wm.preferences, there is an option "disable-workarounds" that for some reason was enabled. I read the description and it's basically meant for workarounds for "Some applications disregard specifications in ways that result in window manager misfeatures. This option puts the WM in a rigorously correct mode, which gives a more consistent user interface, provided one does not need to run any misbehaving applications.". So I disabled it, and it works! 

[Before](http://i.shibe.ml/QmSYKYjzvLKNUmKQBnb8VZ7ybjqBsyReWjfsaNyiUom4Sm.png)

[After](http://i.shibe.ml/QmQPeLioi8B4m5Q8B3YVEN4Mjb7HNUiWYfhYem4fWBByn2.png)

**Comment by mainmachine** [comment]
Congrats @SethDusek ! :+1:

**Comment by SethDusek** [comment]
@mainmachine Damn that was hard to solve though. Your mention of backing up dconf made me first check if there were any settings that could've caused it and that caught my eye. Lucky I figured out what happened otherwise i'd probably have to reconfigure cinnamon again

**Comment by mainmachine** [comment]
That's the only drawback to installing multiple desktops - sometimes they step on each other, luckily not very often. :smile: 

I tried installing KDE Plasma 5 when last year, and it required me to change DM's to SDDM from LightDM, and my Cinnamon and Kodi DE's were totally broken, even after I purged Plasma... of course I may have broken them by trying to "get things to work". This is why we make backups, or practice on the bedroom HTPC. :wink:
